controllers/templatesync/template_sync.go

```diff
@@ -1161,24 +1155,57 @@ func (r *PolicyReconciler) processDependencies(
 			continue
 		}
 
-		// set up namespace for replicated policy dependencies
-		ns := dep.Namespace
-		if ns == "" {
-			ns = r.ClusterNamespace
-		}
+		var res dynamic.ResourceInterface
 
-		// query object and compare compliance status to desired
-		res := dClient.Resource(rsrc).Namespace(ns)
+		if namespaced {
+			ns := dep.Namespace
+			if ns == "" && dep.Group == policiesv1.GroupVersion.Group {
+				// ocm policies should always be in the cluster namespace
+				ns = r.ClusterNamespace
+			}
+
+			res = dClient.Resource(rsrc).Namespace(ns)
+		} else {
+			res = dClient.Resource(rsrc)
+		}
 
 		depObj, err := res.Get(ctx, dep.Name, metav1.GetOptions{})
 		if err != nil {
+			if dep.Group == utils.GvkConstraintTemplate.Group && templateDeps[dep] != "Compliant" {
+				continue // in this (strange) case, the policy wants the ConstraintTemplate to not be found
+			}
+
 			tLogger.Info("Failed to get dependency object", "object", dep)
 
 			dependencyFailures = append(dependencyFailures, dep)
-		} else {
+
+			continue
+		}
+
+		switch dep.Group {
+		case utils.GvkConstraintTemplate.Group:
+			if templateDeps[dep] != "Compliant" {
+				// The ConstraintTemplate was found, but the policy wants it to not be found
+				tLogger.Info("Compliance mismatch for dependency object", "object", dep)
+
+				dependencyFailures = append(dependencyFailures, dep)
+			}
+		case utils.GConstraint:
+			violations, found, err := unstructured.NestedInt64(depObj.Object, "status", "totalViolations")
```

Comment:
I'm thinking that a more generic way rather than duplicate the logic to determine compliance is for any policy template that isn't an OCM policy and is supported, check the compliance according to the replicated root policy. You should be able to find the root policy name by checking the Gatekeeper constraint template/constraint's `policy.open-cluster-management.io/policy` label.

Reply:
That sounds nice, but it shifts some complexity to managing the list of objects to watch: right now I like that the objects a policy watches is just the list of all its dependencies, exactly as they are defined.

The original design was that "dependencies will be able to accept any object that has a status.complianceState field" - technically it isn't necessary for that object to be managed by the policy framework.

Reply:
This is a simple enough check that this is okay.